api/channel.go

```diff
@@ -877,6 +885,8 @@ func updateLastViewedAt(c *Context, w http.ResponseWriter, r *http.Request) {
 		active = true
 	}
 
+	channelName := data["channel_name"].(string)
```

Comment:
This will cause the server to panic on bad input. It should be surrounded by an if statement like the one above for active. I'm 0/5 on if it should error out or provide a default though.